docker/docker-compose.dev.yml

```diff
@@ -0,0 +1,81 @@
+name: lilypad
+services:
+  chain:
+    image: offchainlabs/nitro-node:v2.3.4-b4cc111
+    container_name: chain
+    ports:
+      - 8547:8547
+      - 8548:8548
+    command:
+      [
+        "--init.dev-init",
+        "--init.dev-init-address",
+        $ADMIN_ADDRESS,
+        "--node.dangerous.no-l1-listener",
+        "--node.parent-chain-reader.enable=false",
+        "--parent-chain.id=1337",
+        "--chain.id=412346",
+        "--node.sequencer",
+        "--execution.sequencer.enable",
+        "--node.dangerous.no-sequencer-coordinator",
+        "--node.staker.enable=false",
+        "--init.empty=false",
+        "--http.port",
+        "8547",
+        "--http.addr",
+        "0.0.0.0",
+        "--ws.port",
+        "8548",
+        "--ws.addr",
+        "0.0.0.0",
+      ]
+    volumes:
+      - ../data/chain:/home/user/.arbitrum
+  solver:
+    image: ghcr.io/lilypad-tech/solver
+    container_name: solver
+    depends_on:
+      - chain
+    build:
+      context: ..
+      dockerfile: ./docker/solver/Dockerfile
+      args:
+        - network=dev
+    extra_hosts:
+      - "localhost:host-gateway"
+    environment:
+      - DOPPLER_TOKEN=${DOPPLER_TOKEN_SOLVER}
+    ports:
+      - 8080:8080
+  job-creator:
+    image: ghcr.io/lilypad-tech/job-creator
+    container_name: job-creator
+    depends_on:
+      - solver
+    build:
+      context: ..
+      dockerfile: ./docker/job-creator/Dockerfile
+      args:
+        - network=dev
+    extra_hosts:
+      - "localhost:host-gateway"
+    environment:
+      - DOPPLER_TOKEN=${DOPPLER_TOKEN_JOB_CREATOR}
+  resource-provider:
+    image: ghcr.io/lilypad-tech/resource-provider
+    container_name: resource-provider
+    depends_on:
+      - chain
+      - solver
+    build:
+      context: ..
+      dockerfile: ./docker/resource-provider/Dockerfile
+      args:
+        - NETWORK=dev
+    extra_hosts:
+      - "localhost:host-gateway"
+    volumes:
+      - ../data/bacalhau:/root/.bacalhau
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      - WEB3_PRIVATE_KEY=0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a
```

Comment:
Any chance we can pull this in from `doppler`?

Reply:
Oh good catch! Yes, I'll update

Reply:
Follow up question: which key is this one? and why do you need it around? Each service has its own keys injected via `doppler` (even `run` and `onchain`), so why do we need another key in the `env`?

Reply:
The RP docker image doesn't use doppler (because RPs won't have access to our doppler?) - but i've wrapped the `docker compose up` with doppler run to inject it for our use.